### PR TITLE
28 create a new interface iconvergeonepicschannelaccess for handling all epics channel access implementations

### LINKED ISSUE #28: Create a new ConvergeOnEPICSChannelAccess class for handling all EPICS channel hub instances

### PR DESCRIPTION
**Pull request by TineZata**
@aoun-muhammad ... when you are free have a look at this pull request.

### CODE REVIEW

File: Convergence/IO/EPICS/CA/ChannelAccessWrapper.cs [hunk 1]
```diff
@@ -82,15 +78,15 @@ static ChannelAccessDLLWrapper()
...
 
-        public static IntPtr CurrentContext { get; private set; }
+        public static nint CurrentContext { get; private set; }
 
         private static void set_current_context()
         {
             var context = ca_current_context();
             // Check if CurrentContext has been created
-            if (CurrentContext == IntPtr.Zero)
+            if (CurrentContext == nint.Zero)
```
Comment (aoun-muhammad) [inline Convergence/IO/EPICS/CA/ChannelAccessWrapper.cs:87]: I liked that you are using modern C# datatypes 😃

File: Convergence/IO/EPICS/CA/ChannelAccessWrapper.cs [hunk 2]
```diff
@@ -149,26 +145,26 @@ public static EcaType ca_context_create(PreemptiveCallbacks preemptive)
...
-            static extern IntPtr ca_current_context();
+            static extern nint ca_current_context();
         }
 
-        public static EcaType ca_attach_context(this IntPtr context)
+        public static EcaType ca_attach_context(this nint context)
         {
             // Try Parse Enum of type <EcaType> from Int32 return by (ca_attach_context(context))
-            if (Enum.TryParse<EcaType>(CA_EXTRACT_MSG_NO(ca_attach_context(context)).ToString(), out EcaType result))
+            if (Enum.TryParse(CA_EXTRACT_MSG_NO(ca_attach_context(context)).ToString(), out EcaType result))
```
Comment (aoun-muhammad) [inline Convergence/IO/EPICS/CA/ChannelAccessWrapper.cs:167]: Any specific reason to remove <EcaType>?
I guess it just provide an extra check.

File: Convergence/IO/EPICS/CA/ChannelHandle.cs [hunk 3]
```diff
@@ -0,0 +1,35 @@
+﻿namespace Convergence.IO.EPICS.CA
```
Comment (aoun-muhammad) [inline Convergence/IO/EPICS/CA/ChannelHandle.cs]: Why you need a separate ChannelHandle for just CS?

Can't this be generic? where a single ChannelHandle should cover any type of IO?

**Review by aoun-muhammad — APPROVED** [review]

**Comment by aoun-muhammad** [comment]
@TineZata I have 2-3 questions which I have put in as comments.
Apart from that it looks okay.
I am happy to merge it.

File: Convergence/IO/EPICS/CA/ChannelHandle.cs (hunk 3 above)
Comment (TineZata) [reply Convergence/IO/EPICS/CA/ChannelHandle.cs]: I missed this... its not doing anything. I will delete it.

File: Convergence/IO/EPICS/CA/ChannelAccessWrapper.cs (hunk 1 above)
Comment (TineZata) [reply Convergence/IO/EPICS/CA/ChannelAccessWrapper.cs:87]: Actually... not very consistently ... maybe I will just replace all IntPtr with nint

**Comment by TineZata** [comment]
> @TineZata I have 2-3 questions which I have put in as comments. Apart from that it looks okay. I am happy to merge it.

Thanks @aoun-muhammad ... i've made the changes you suggested.

Good suggestions 👍

**Pull request merged**